docs/src/main/asciidoc/datasource.adoc

```diff
@@ -517,6 +517,72 @@ public class MyProducer {
 ----
 ====
 
+[[datasource-multiple-single-transaction]]
+=== Use multiple datasources in a single transaction
+
+By default, a transaction may include at most one datasource.
+Attempting to access multiple datasources in the same transaction
+would result in an exception similar to this:
+
+[source]
+----
+...
+Caused by: java.sql.SQLException: Exception in association of connection to existing transaction
+        at io.agroal.narayana.NarayanaTransactionIntegration.associate(NarayanaTransactionIntegration.java:130)
+        ...
+Caused by: java.sql.SQLException: Unable to enlist connection to existing transaction
+        at io.agroal.narayana.NarayanaTransactionIntegration.associate(NarayanaTransactionIntegration.java:121)
+        ...
+----
+
+To allow using multiple JDBC datasources in the same transaction:
+
+. Make sure your JDBC driver supports XA.
+All <<extensions-and-database-drivers-reference,supported JDBC drivers do>>,
+but <<other-databases,other JDBC drivers>> might not.
+. Make sure your database server is configured to enable XA.
+. Enable XA support explicitly for each relevant datasource by setting
+<<quarkus-agroal_quarkus-datasource-jdbc-transactions,`quarkus.datasource[.optional name].transactions`>> to `xa`.
+
+Using XA, a rollback in one datasource will trigger a rollback in every other datasource enrolled in the transaction.
+
+[NOTE]
+====
+XA transactions on reactive datasources are not supported at the moment.
+====
+
+If XA cannot be enabled for one of your datasources:
+
+* Be aware that enabling XA for all datasources _except one_ (and only one) is still supported
+through https://www.narayana.io/docs/project/index.html#d5e857[Last Resource Commit Optimization (LRCO)],
+though some errors may cause an inconsistent transaction outcome.
```

Comment:
I think you're referring to the cause of an heuristic outcome here, but this is possible with both LRCO and a regular couple of XA resources. Perhaps this sounds a little too scary as LRCO is generally in the same ballpark of risk as XA? 

There are two options here - either one is very precise about which errors you're referring to exactly so to not have a hand-wavy scary tone, or one could keep it simple and not get into such details.

I'd recommend to keep it simple and just remove this last sentence, especially as you've already linked to the Narayana docs.

@mmusgrov , WDYT?